src/execution.rs

```diff
@@ -689,3 +686,39 @@ impl UnscopedVariable {
         })
     }
 }
+
+/// Get the value for the given capture, considering the suffix
+pub fn query_capture_value<'tree>(
+    index: usize,
+    suffix: QueryCaptureSuffix,
+    mat: &QueryMatch<'_, 'tree>,
+    graph: &mut Graph<'tree>,
+) -> Value {
+    let nodes = mat
+        .captures
+        .iter()
+        .filter(|c| c.index as usize == index)
+        .map(|c| c.node)
+        .collect::<Vec<_>>();
+    match suffix {
+        QueryCaptureSuffix::One => {
+            let syntax_node = graph.add_syntax_node(nodes[0]);
+            syntax_node.into()
+        }
+        QueryCaptureSuffix::ZeroOrMore | QueryCaptureSuffix::OneOrMore => {
+            let syntax_nodes = nodes
+                .iter()
+                .map(|n| graph.add_syntax_node(n.clone()).into())
+                .collect::<Vec<Value>>();
+            syntax_nodes.into()
+        }
```

Comment:
What happens if an outer `+` or `*` capture contains inner captures?
E.g. something like
```
[
    (foo (_) @bar)
    (baz (_) @quux)
]+ @foo_or_baz
```
Would `@bar`/`@quux` just fail to to have a value at all? Or would it end up with the value corresponding to the last capture of `@foo_or_baz`?

Reply:
Good question. I expect it will have the last value, but I haven't tested that yet.

Reply:
One possibility (though I don't know how easy it would be be to implement) would be to also produce lists of captures for `@bar` and `@quux`, but use `#null` (or whatever we use to indicate "no capture") for those indices where there was no capture.

I don't know how useful this would be, though. To tie, say `@bar` together with `@foo_or_baz`, you would then have to zip together these two lists.

Reply:
Feedback on tree-sitter/tree-sitter#1504 noted that my approach was too simplistic, because it doesn't account for these nesting scenarios. I am revisiting that to make sure that is handled better.

`tree-sitter` outputs a list of captured nodes, indicating to which capture the node belongs. The output will be simpler than a full matching tree, so reconstructing such rich output that allows us to match nodes of one capture to nodes of another would require (probably invasive) tree-sitter changes. But I also think that this would over-complicate things, without clear benefits.

Consider the following example:

    (foo (_)? @bar)* @foos
    (bar (_)? @baz)+ @bars

The quantifier of `@bar` would be `*` (combining `*` and `?`), and the value a list. The quantifier for `@baz` would also be `*` (combining `+` and `?`).

If logic is necessary to tie one match to another, for example an element of `@foos` with the corresponding element in `@bar`, this can always be expressed by writing a different stanza that matches `(foor (_)? @bar) @foo`.